Stratis.Bitcoin.Features.Consensus/PosConsensusValidator.cs

```diff
@@ -448,7 +450,12 @@ public void CheckAndComputeStake(ContextInformation context)
                 if (prevBlockStake == null)
                     ConsensusErrors.PrevStakeNull.Throw();
 
-                this.stakeValidator.CheckProofOfStake(context, pindexPrev, prevBlockStake, block.Transactions[1], pindex.Header.Bits.ToCompact());
+                // Only do proof of stake validation for blocks after last checkpoint.
+                if (pindex.Height > lastCheckpointHeight)
```

Comment:
A side effect is we will not have the stake information of all the other blocks, this needs to be configurable in my opinion.

Reply:
I am not sure I follow here. I understand the general requirement of having this configurable, this is quite standard requirement for users who do not want to trust devs that much. But I do not understand the side effect you mention.

Reply:
I will create issue for adding checkpoint config switch.